scalafix-reflect/src/main/scala-3/scalafix/internal/reflect/RuleCompiler.scala

```diff
@@ -0,0 +1,88 @@
+package scalafix.internal.reflect
+
+import java.io.File
+import java.nio.file.Paths
+import scala.meta.io.AbsolutePath
+
+import dotty.tools.dotc.Compiler
+import dotty.tools.dotc.Run
+import dotty.tools.dotc.util.SourceFile
+import dotty.tools.dotc.reporting.StoreReporter
+import dotty.tools.io.AbstractFile as DottyAbstractFile
+
+import scala.reflect.io.VirtualDirectory
+import scala.reflect.io.AbstractFile
+import scala.reflect.internal.util.AbstractFileClassLoader
+
+import metaconfig.Configured
+import metaconfig.Input
+import metaconfig.ConfError
+import metaconfig.Position
+import dotty.tools.dotc.config.Settings.Setting
+import dotty.tools.dotc.config.Settings.ArgsSummary
+import dotty.tools.dotc.core.Contexts.ContextBase
+import dotty.tools.dotc.interactive.InteractiveDriver
+import dotty.tools.dotc.core.Contexts.FreshContext
+
+class RuleCompiler(
+    classpath: String,
+    target: AbstractFile = new VirtualDirectory("(memory)", None)
+) {
+  // we are not configuring any settings!
+  // private val settings = new Settings()
+  // settings.deprecation.value = true // enable detailed deprecation warnings
+  // settings.unchecked.value = true // enable detailed unchecked warnings
+  // settings.outputDirs.setSingleOutput(target)
+  // settings.classpath.value = classpath
+  private val settings = "-unchecked" :: "-deprecation" :: Nil
+  private val driver = new InteractiveDriver(settings.distinct)
+  private val reporter: StoreReporter = new StoreReporter()
+  private given ctx: FreshContext = driver.currentCtx.fresh
+
+  ctx
+    .setReporter(reporter)
+    .setSetting(ctx.settings.outputDir, DottyAbstractFile.getFile(target.path))
+    .setSetting(ctx.settings.classpath, classpath)
```

Comment:
I've found examples of dotty usage from which I've implemented the setup of the context and later the usage of the compiler / run for scala3:
- [scalameta](https://github.com/scalameta/mdoc/blob/c52156730d3bdd485dd213b9d195402bbe255174/mdoc/src/main/scala-3/mdoc/internal/markdown/MarkdownCompiler.scala#L56)
- [scala compiler-benchmark](https://github.com/scala/compiler-benchmark/blob/90f1b17f570bd66562a473f9d3de3f698e0fde28/compilation/src/main/dotc/scala/tools/benchmark/BenchmarkDriver.scala)
- [dotty SnippetCompiler](https://github.com/lampepfl/dotty/blob/8bd527ed54bed91294373f7e20ba73565b9324ff/scaladoc/src/dotty/tools/scaladoc/snippets/SnippetCompiler.scala)

Now it looks more reasonable

Reply:
are you sure this has an effect? by having a quick look at https://github.com/scalameta/mdoc/blob/c52156730d3bdd485dd213b9d195402bbe255174/mdoc/src/main/scala-3/mdoc/internal/markdown/MarkdownCompiler.scala#L56-L72, it seems that `ctx` is immutable and `setSetting` is returning a copy (discarded above)

Reply:
indeed it returns a new context, I changed the implementation and reassign the context